src/gt4py/next/program_processors/runners/dace_fieldview/transformations/auto_optimize.py

```diff
@@ -20,146 +20,12 @@
-        skip=(skip | {"InlineSDFGs"}),
-    ).apply_pass(sdfg, {})
-
-
-def gt_set_iteration_order(
-    sdfg: dace.SDFG,
-    leading_dim: gtx_common.Dimension,
-    validate: bool = True,
-    validate_all: bool = False,
-) -> Any:
-    """Set the iteration order of the Maps correctly.
-
-    Modifies the order of the Map parameters such that `leading_dim`
-    is the fastest varying one, the order of the other dimensions in
-    a Map is unspecific. `leading_dim` should be the dimensions were
-    the stride is one.
-
-    Args:
-        sdfg: The SDFG to process.
-        leading_dim: The leading dimensions.
-        validate: Perform validation during the steps.
-        validate_all: Perform extensive validation.
-    """
-    return sdfg.apply_transformations_once_everywhere(
-        gtx_transformations.MapIterationOrder(
-            leading_dim=leading_dim,
-        ),
-        validate=validate,
-        validate_all=validate_all,
-    )
-
-
-def gt_inline_nested_sdfg(
-    sdfg: dace.SDFG,
-    multistate: bool = True,
-    permissive: bool = False,
-    validate: bool = True,
-    validate_all: bool = False,
-) -> dace.SDFG:
-    """Perform inlining of nested SDFG into their parent SDFG.
-
-    The function uses DaCe's `InlineSDFG` transformation, the same used in simplify.
-    However, before the inline transformation is run the function will run some
-    cleaning passes that allows inlining nested SDFGs.
-    As a side effect, the function will split stages into more states.
-
-    Args:
-        sdfg: The SDFG that should be processed, will be modified in place and returned.
-        multistate: Allow inlining of multistate nested SDFG, defaults to `True`.
-        permissive: Be less strict on the accepted SDFGs.
-        validate: Perform validation after the transformation has finished.
-        validate_all: Performs extensive validation.
-    """
-    first_iteration = True
-    i = 0
-    while True:
-        print(f"ITERATION: {i}")
-        nb_preproccess = sdfg.apply_transformations_repeated(
-            [dace_dataflow.PruneSymbols, dace_dataflow.PruneConnectors],
-            validate=False,
-            validate_all=validate_all,
-        )
-        if (nb_preproccess == 0) and (not first_iteration):
-            break
-
-        # Create and configure the inline pass
-        inline_sdfg = dace_passes.InlineSDFGs()
-        inline_sdfg.progress = False
-        inline_sdfg.permissive = permissive
-        inline_sdfg.multistate = multistate
-
-        # Apply the inline pass
-        nb_inlines = inline_sdfg.apply_pass(sdfg, {})
-
-        # Check result, if needed and test if we can stop
-        if validate_all or validate:
-            sdfg.validate()
-        if nb_inlines == 0:
-            break
-        first_iteration = False
-
-    return sdfg
-
-
 def gt_auto_optimize(
     sdfg: dace.SDFG,
     gpu: bool,
-    leading_dim: Optional[gtx_common.Dimension] = None,
+    leading_dim: Optional[
```

Comment:
For TaskletFusion the Tasklets need to be in the same scope, which means Map in DaCe.
Since all itermediate are registers anyway the compiler can do that anyway.
You should once look at the assembly code that is generated it is quite impressive.

What you propose, is a highly specialized transformation, that should be solved through a library node.

Reply:
I was thinking that maybe DaCe runs analysis on the Tasklet code, and if tasklets are fused then "something more" could be analyzed and possibly simplified.

Reply:
However, we could try it separately.